<?xml version="1.0"?>
<!--
Version:  2017.07.31
Authors:  Members of the Team NAIST-Panasonic at the Amazon Robotics Challenge 2017
          Gustavo A. Garcia R. garcia-g at is.naist.jp (Captain), 
          Lotfi El Hafi, Felix von Drigalski, Wataru Yamazaki, Viktor Hoerig, Arnaud Delmotte, 
          Akishige Yuguchi, Marcus Gall, Chika Shiogama, Kenta Toyoshima, Pedro Uriguen, 
          Rodrigo Elizalde, Masaki Yamamoto, Yasunao Okazaki, Kazuo Inoue, Katsuhiko Asai, 
          Ryutaro Futakuchi, Seigo Okada, Yusuke Kato, and Pin-Chu Yang
*******************
Copyright 2017 Team NAIST-Panasonic 
Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at 
    http://www.apache.org/licenses/LICENSE-2.0 
Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.
*******************
-->

<launch>
  <!-- Name of the tool we are going to use -->
  <arg name="tool_name" default="tnptool"/> 
  
  <!-- Parameters to pass to the ROS node -->
  <arg name="move_group" default="manipulator"/>
  <arg name="ee_link" default="tnptool_suction_link_ee"/>
  <arg name="ros_rate" default="0.1"/>
  
  <param name="/iiwa/toolName" type="string" value="$(arg tool_name)" /> <!--Sunrise tool-->
  
  <!--  This loads the robot description  -->
  <include file="$(find iiwa_tnptool_moveit)/launch/moveit_planning_execution.launch">
    <arg name="sim" value="false"/>
  </include>
    
  <!-- Here we call the ROS node we have written, with the parameters on top   -->
  <node ns="iiwa" name="kinematics_trial" pkg="tnp_moveit_planner" type="kinematics_trial" respawn="false" output="screen">
    <param name="move_group" value="$(arg move_group)"/>
    <param name="ee_link" value="$(arg ee_link)"/>
    <param name="ros_rate" value="$(arg ros_rate)"/>
  </node>

</launch>